Mathlib/Analysis/Calculus/FDeriv/Analytic.lean

```diff
@@ -200,3 +178,117 @@ theorem AnalyticOn.iterated_deriv [CompleteSpace F] (h : AnalyticOn 𝕜 f s) (n
 #align analytic_on.iterated_deriv AnalyticOn.iterated_deriv
 
 end deriv
+section fderiv
+
+variable {p : FormalMultilinearSeries 𝕜 E F} {r : ℝ≥0∞} {n : ℕ}
+
+variable {f : E → F} {x : E} {s : Set E}
+
+/-! The case of continuously polynomial functions. We get the same differentiability
+results as for analytic functions, but without the assumptions that `F` is complete.-/
+
+theorem HasFiniteFPowerSeriesOnBall.differentiableOn
+    (h : HasFiniteFPowerSeriesOnBall f p x n r) : DifferentiableOn 𝕜 f (EMetric.ball x r) :=
+  fun _ hy ↦ (h.cPolynomialAt_of_mem hy).analyticAt.differentiableWithinAt
+
+theorem HasFiniteFPowerSeriesOnBall.hasFDerivAt (h : HasFiniteFPowerSeriesOnBall f p x n r)
+    {y : E} (hy : (‖y‖₊ : ℝ≥0∞) < r) :
+    HasFDerivAt f (continuousMultilinearCurryFin1 𝕜 E F (p.changeOrigin y 1)) (x + y) :=
+  (h.changeOrigin hy).toHasFPowerSeriesOnBall.hasFPowerSeriesAt.hasFDerivAt
+
+theorem HasFiniteFPowerSeriesOnBall.fderiv_eq (h : HasFiniteFPowerSeriesOnBall f p x n r)
+    {y : E} (hy : (‖y‖₊ : ℝ≥0∞) < r) :
+    fderiv 𝕜 f (x + y) = continuousMultilinearCurryFin1 𝕜 E F (p.changeOrigin y 1) :=
+  (h.hasFDerivAt hy).fderiv
+
+/-- If a function has a finite power series on a ball, then so does its derivative. -/
+protected theorem HasFiniteFPowerSeriesOnBall.fderiv
+    (h : HasFiniteFPowerSeriesOnBall f p x (n + 1) r) :
+    HasFiniteFPowerSeriesOnBall (fderiv 𝕜 f)
+      ((continuousMultilinearCurryFin1 𝕜 E F : (E[×1]→L[𝕜] F) →L[𝕜] E →L[𝕜] F)
+        |>.compFormalMultilinearSeries (p.changeOriginSeries 1)) x n r := by
+  refine .congr (f := fun z ↦ continuousMultilinearCurryFin1 𝕜 E F (p.changeOrigin (z - x) 1)) ?_
+    fun z hz ↦ ?_
+  · refine continuousMultilinearCurryFin1 𝕜 E F
+      |>.toContinuousLinearEquiv.toContinuousLinearMap.comp_hasFiniteFPowerSeriesOnBall ?_
+    simpa using
+      ((p.hasFiniteFPowerSeriesOnBall_changeOrigin 1 h.finite).mono h.r_pos le_top).comp_sub x
+  dsimp only
+  rw [← h.fderiv_eq, add_sub_cancel'_right]
+  simpa only [edist_eq_coe_nnnorm_sub, EMetric.mem_ball] using hz
+
+/-- Variant of the previous result where the degree of `f` is `< n` and not `< n + 1`. -/
```

Comment:
"previous result" doesn't make sense when the docstring is displayed in a tooltip outside this file. Please give an absolute reference.